src/sage/graphs/generators/families.py

```diff
@@ -563,17 +557,41 @@ def BalancedTree(r, h):
     has degree `r \geq 2`, but the construction degenerates
     gracefully::
 
-        sage: graphs.BalancedTree(1, 10)                                                # needs networkx
+        sage: graphs.BalancedTree(1, 10)
         Balanced tree: Graph on 11 vertices
 
     Similarly, we usually want the tree must have height `h \geq 1`
     but the algorithm also degenerates gracefully here::
 
-        sage: graphs.BalancedTree(3, 0)                                                 # needs networkx
+        sage: graphs.BalancedTree(3, 0)
         Balanced tree: Graph on 1 vertex
+
+    The construction is the same as the one of networkx::
+
+        sage: # needs networkx
+        sage: import networkx
+        sage: r = randint(2, 4); h = randint(1, 5)
+        sage: T = graphs.BalancedTree(r, h)
+        sage: N = Graph(networkx.balanced_tree(r, h), name="Balanced tree")
+        sage: T.is_isomorphic(N)
+        True
     """
-    import networkx
-    return Graph(networkx.balanced_tree(r, h), name="Balanced tree")
+    # Compute the number of vertices per level of the tree
+    order = [r**l for l in range(h + 1)]
+    # Compute the first index of the vertices of a level
+    begin = [0]
+    for level in range(1, h + 2):
+        begin.append(begin[level - 1] + order[level - 1])
```

Comment:
```suggestion
    for level in range(h + 1):
        begin.append(begin[level] + order[level]) # level is 0-based
```
Micro-optimization. Although you could do
```python
    begin.extend(begin[-1] + val for val in order)
```
but that is more subtle.

Reply:
I like it.